### PR TITLE
New signup login, remove user object

### PR DESCRIPTION
**Pull request by Steven-Chan**
connect #142

### CODE REVIEW

**Comment by Steven-Chan** [comment]
@ben181231 ready for review

File: skygear/src/main/java/io/skygear/skygear/PersistentStore.java [hunk 1]
```diff
@@ -102,26 +108,38 @@ void save() {
...
         if (this.currentUser != null) {
             prefEditor.putString(CURRENT_USER_KEY,
-                    UserSerializer.serialize(this.currentUser).toString()
+                    RecordSerializer.serialize(this.currentUser).toString()
             );
         } else {
             prefEditor.remove(CURRENT_USER_KEY);
         }
+
+        if (this.accessToken != null) {
```
Comment (ben181231) [inline skygear/src/main/java/io/skygear/skygear/PersistentStore.java]: Move this line to a new method `private void saveAccessToken(SharedPreferences.Editor prefEditor)`, following the pattern of this class.

File: skygear/src/main/java/io/skygear/skygear/AssignUserRoleRequest.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        String[] roleNames = new String[roles.length];
+        for (int idx = 0; idx < roles.length; idx++) {
+            roleNames[idx] = roles[idx].getName();
+        }
+
+        this.data.put("users", userIDs);
+        this.data.put("roles", roleNames);
+    }
+
+    private void validateUserRecords(Record[] users) {
```
Comment (ben181231) [inline skygear/src/main/java/io/skygear/skygear/AssignUserRoleRequest.java]: Can we move this logic to a overridden method `protected void validate() throws Exception` ?

File: skygear/src/main/java/io/skygear/skygear/GetUserRoleRequest.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        String[] userIDs = new String[users.length];
+        for (int idx = 0; idx < users.length; idx++) {
+            userIDs[idx] = users[idx].getId();
+        }
+
+        this.data.put("users", userIDs);
+    }
+
+    private void validateUserRecords(Record[] users) {
```
Comment (ben181231) [inline skygear/src/main/java/io/skygear/skygear/GetUserRoleRequest.java]: Can we move this logic to a overridden method `protected void validate() throws Exception` ?

File: skygear/src/main/java/io/skygear/skygear/AssignUserRoleRequest.java (hunk 2 above)
Comment (Steven-Chan) [inline skygear/src/main/java/io/skygear/skygear/AssignUserRoleRequest.java]: But the `validate` method is validating the request data, unless the `users` is a state of the request object, I think it makes more sense to be put in where the `users` is passed in to the request object.

Comment (ben181231) [inline skygear/src/main/java/io/skygear/skygear/AssignUserRoleRequest.java]: I think `validate()` is validating the request (including data and state). Maybe keeping `users` as a state for validation is okay.

**Comment by Steven-Chan** [comment]
@ben181231 updated with the rename

**Pull request merged**